truss-chains/truss_chains/deployment/deployment_client.py

```diff
@@ -775,3 +864,20 @@ def watch(
         included_chainlets,
     )
     patcher.watch()
+
+
+def watch_model(
+    source: pathlib.Path,
+    model_name: str,
+    remote_provider: b10_remote.TrussRemote,
+    console: "rich_console.Console",
+    error_console: "rich_console.Console",
+):
+    patcher = _ModelWatcher(
+        source=source,
+        model_name=model_name,
+        remote_provider=cast(b10_remote.BasetenRemote, remote_provider),
```

Comment:
I don't love the explicit cast, but the interface method `sync_truss_to_dev_version_by_name` doesn't quite work for our use case with different directories for source / generated truss. Therefore, we need to do something similar to chainlets and peek into more low level `patch` API